### ISSUE TITLE
MD076: Inconsistent list item spacing not detected when code block inside list item

### ISSUE THREAD

**Issue by sisp** [rvben/rumdl#500]
### rumdl version

0.1.42

### Installation method

Pre-built binary

### Bug description

A code block inside a list item seems to reset the `MD076` rule check for consistent list item spacing.

### Steps to reproduce

1. Create a file `test.md` with this content:

    ```markdown
    # Test

    - Lorem ipsum dolor sit amet.
    - Lorem ipsum dolor sit amet.

        ```yaml
        hello: world
        ```

    - Lorem ipsum dolor sit amet.

    - Lorem ipsum dolor sit amet.
    ```

1. Run `rumdl check test.md`

1. See _no_ linter error despite the inconsistent list item spacing.

### Expected behavior

There should be a linter error about inconsistent list item spacing.

### Actual behavior

There is no linter error about inconsistent list item spacing.

### Configuration

```toml
# The default config
```

### Operating system

Linux

### Additional context

Removing the code block inside the list item

```diff
 # Test

 - Lorem ipsum dolor sit amet.
 - Lorem ipsum dolor sit amet.
-
-    ```yaml
-    hello: world
-    ```

 - Lorem ipsum dolor sit amet.

 - Lorem ipsum dolor sit amet.
```

results in a linter error:

```console
$ rumdl check test.md 
test.md:4:1: [MD076] Missing blank line between list items

Issues: Found 1 issues in 1 file (8ms)
```

Also, other markup like a blockquote inside the list item

```diff
 # Test

 - Lorem ipsum dolor sit amet.
 - Lorem ipsum dolor sit amet.

-    ```yaml
-    hello: world
-    ```
+    > Lorem ipsum dolor sit amet.

 - Lorem ipsum dolor sit amet.

 - Lorem ipsum dolor sit amet.
```

results in a linter error:

```console
$ rumdl check test.md 
test.md:4:1: [MD076] Missing blank line between list items

Issues: Found 1 issues in 1 file (8ms)
```

So far, I've only observed this behavior with a code block.

**Comment by rvben** [comment]
Thanks for reporting this, @sisp. Fixed in v0.1.43 — MD076 now correctly detects inconsistent list item spacing when code blocks are inside list items. Closing as resolved.

**Comment by sisp** [comment]
Thank you very much for the super quick fix, @rvben! :bow:
